src/components/svgSetting/svgSetting.js

```diff
@@ -1,34 +1,109 @@
-import React, { Component } from 'react';
-import '../checkBoxSelection/checkBoxSelection.scss';
-import PropTypes from 'prop-types';
+import React, { Component } from "react";
+import "../checkBoxSelection/checkBoxSelection.scss";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { Svgo } from "../svgSetting/svgoConfig";
+import {
+  ADD_VARIABLE,
+  UPDATE_VARIABLE,
+  DELETE_VARIABLE
+} from "../../store/actionTypes";
+
 class SvgSetting extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isClicked: this.props.isSelected
-        };
+  constructor(props) {
+    super(props);
+    this.state = {
+      isClicked: this.props.isSelected,
+      arr: [],
+      svgoObject: { ...Svgo },
+      isFound: false
+    };
+  }
+
+  handleDivClick = () => {
+    this.setState({
+      isClicked: !this.state.isClicked
+    });
+    for (var file in this.props.svgOptions) {
+      if (this.props.svgOptions[file] === this.props.value) {
+        this.state.isFound = true;
+        this.state.index = file;
+      }
+    }
+    if (!this.state.isFound) {
+      this.props.svgOptions.push(this.props.value);
+    } else {
+      this.props.svgOptions.splice(this.state.index, 1);
+      this.state.isFound = false;
     }
+  };
+
+  toChangePluginObjectValues = arrayWithPluginOptions => {
+    let newSvgoObject = this.props.svgObject.plugins.filter(
+      option => {
+        for (var element of arrayWithPluginOptions) {
+          if (option[element] !== null && option[element] !== undefined) {
+            option[element] = true;
+          }
+          return true;
```

Comment:
would this not exit the loop on the first itteration

Reply:
is that intended

Reply:
So you intend on exiting the second loop after 1 iteration every time ?

Reply:
also, correct me if i'm wrong, but doesnt return exit at the function level and not at block level? meaning return will end all execution within the function and return a value. Meaning it would exit the first loop as well

Reply:
I went through that and also got confused some how , trying to make sense out of it

Reply:
Yhea, you are correct its `break` that exit at block level and `return` exit at function level. I will update this function and test any abnormalities.

Reply:
what a waste of our time this function is not even used anywhere hence it does not make sense, Apologies for your time too.

Reply:
lollllll, ok well atleast we learnt some things along the way